### PR TITLE
[cleanup] Remove default index on dims.

### PR DESCRIPTION
**Pull request by rdzhabarov**
This is the followup from the https://github.com/pytorch/glow/pull/1224

Need to check OpenCL backend as well (looks good).

### CODE REVIEW

**Comment by qcolombet** [comment]
Did you expose any bug while doing this?
If so, I am curious what we could have done to catch them earlier.

**Comment by rdzhabarov** [comment]
no bugs, but there were some bugs with the getType() with the default index exposed earlier :)
those bugs would be caught during debug build if the network exposes certain structure, but it's tedious every time to fix those places. So, I went ahead and fixed all those now.

**Comment by qcolombet** [comment]
> those bugs would be caught with the debug build run if the network exposes certain structure, but it's tedious every time to fix those places. So, I went ahead and fixed all those now.

Sounds good. I was mostly wondering if we could add some tests that would have catch the issues (same for getType).

File: lib/Graph/Node.cpp [hunk 1]
```diff
@@ -80,11 +80,6 @@ void Node::setPredicate(const NodeValue &P) { predicate_ = P; }
 bool Node::hasPredicate() const { return predicate_.getNode(); }
 
 TypeRef Node::getType(unsigned idx) const {
-  if (idx == (unsigned)-1) {
-    assert(numRes_ == 1 && "Did not specify the result number for a node "
-                           "with multiple results.");
-    return types_[0];
-  }
```
Comment (qcolombet) [inline lib/Graph/Node.cpp]: That's the code I was talking about :).
Thanks!

Out-of-curiosity, do you know why the coverage bot, reports this as covered? (That was what I was checking when I asked you about our coverage checking.)

File: lib/Optimizer/GraphOptimizer.cpp [hunk 2]
```diff
@@ -1492,7 +1492,7 @@ static bool sinkRescaleQuantizedNode(Function *F) {
       }
 
       auto *newReshape = F->createReshape(reshape->getName(),
-                                          rescale->getInput(), reshape->dims());
+                                          rescale->getInput(), reshape->getResult().dims());
```
Comment (jfix71) [inline lib/Optimizer/GraphOptimizer.cpp]: nit: make sure you run clang-format before merging

**Comment by qcolombet** [comment]
> I'm wondering if we should change all the get/create Variables functions in the proto loaders to return Variable instead of Node.

We create loads now, so no we can’t + in theory went you call get you can get a random node (whatever was registered under that name)

**Comment by qcolombet** [comment]
@rdzhabarov correct!

File: lib/Graph/Node.cpp (hunk 1 above)
Comment (rdzhabarov) [reply lib/Graph/Node.cpp]: Not sure, will look into this.

File: lib/Optimizer/GraphOptimizer.cpp (hunk 2 above)
Comment (rdzhabarov) [reply lib/Optimizer/GraphOptimizer.cpp]: damn :( forgot! will run real quick.

Comment (rdzhabarov) [reply lib/Optimizer/GraphOptimizer.cpp]: we need to have a clang format checker before merge :) (seperate story).

https://github.com/pytorch/glow/pull/1253

**Pull request merged**